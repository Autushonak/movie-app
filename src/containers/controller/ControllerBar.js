import React, { useState } from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import { useRouteMatch } from 'react-router-dom';
import { colors } from '../../constants/colors';
import ButtonGroup from '../button-group/ButtonGroup';
import { ResultsTitle } from '../../components';

const ControllerBarStyled = styled.div`
  display: flex;
  flex-direction: row;
  background-color: ${colors.GREY_0};
  justify-content: space-between;
  padding: 0 150px;
`;

const MainPageControllerBar = () => {
  const [sortBy, setSortBy] = useState('release date');
  return (
    <>
      <ResultsTitle>7 movie found</ResultsTitle>
      <ButtonGroup
        title="Sort by"
        values={['release date', 'rating']}
        activeBtn={sortBy}
        onClick={setSortBy}
      ></ButtonGroup>
    </>
  );
};

const MoviePageControllerBar = () => <ResultsTitle>Films By Drama Genre</ResultsTitle>;

const ControllerBar = () => {
  const match = useRouteMatch('/movie');
  return <ControllerBarStyled>{match ? <MoviePageControllerBar /> : <MainPageControllerBar />}</ControllerBarStyled>;
};

ControllerBar.propTypes = {
  match: PropTypes.object,
};

export default ControllerBar;
