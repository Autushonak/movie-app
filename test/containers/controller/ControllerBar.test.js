import 'jest-styled-components';
import React from 'react';
import { shallow } from 'enzyme';
import ControllerBar, { ControllerBarStyled } from '../../../src/containers/controller/ControllerBar';

jest.mock('react-router-dom', () => ({
  useRouteMatch: () => true,
}));

describe('ControllerBarStyled', () => {
  test('should be rendered correctly', () => {
    const component = shallow(<ControllerBarStyled />);
    expect(component).toMatchSnapshot();
  });
});

describe('ControllerBar', () => {
  test('should be rendered correctly for movie page', () => {
    const component = shallow(<ControllerBar />);
    expect(component).toMatchSnapshot();
  });
});
