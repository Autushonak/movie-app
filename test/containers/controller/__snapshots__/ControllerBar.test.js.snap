// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ControllerBar should be rendered correctly for main page 1`] = `
<styled.div>
  <MainPageControllerBar />
</styled.div>
`;

exports[`ControllerBar should be rendered correctly for movie page 1`] = `
<styled.div>
  <MoviePageControllerBar />
</styled.div>
`;

exports[`ControllerBarStyled should be rendered correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  background-color: #555555;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding: 0 150px;
}

<div
  className="c0"
/>
`;
